commit f90446aec14f82d223c22f13efec17752b6ed7bb
Author: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date:   Fri Nov 3 23:27:57 2017 -0400

    Wait longer for test lfs server to start.
    
    Also, back off on polling a bit after the first check to reduce any I/O
    contention that might slow things down even more. Instead of 5 seconds,
    this will wait up to about 60 seconds for files created by the test
    server to appear. This allows very slow systems to be able to pass
    tests.
    
    Finally, add a message if the file doesn't appear to make it clear what
    went wrong.

Patch-Source: https://src.fedoraproject.org/rpms/git-lfs/blob/f27/f/wait-longer.patch

diff --git a/test/testhelpers.sh b/test/testhelpers.sh
index 531d934b..cb31d2ef 100644
--- a/test/testhelpers.sh
+++ b/test/testhelpers.sh
@@ -258,15 +258,20 @@ size %s
 wait_for_file() {
   local filename="$1"
   n=0
-  while [ $n -lt 10 ]; do
+  wait_time=1
+  while [ $n -lt 17 ]; do
     if [ -s $filename ]; then
       return 0
     fi
 
-    sleep 0.5
+    sleep $wait_time
     n=`expr $n + 1`
+    if [ $wait_time -lt 4 ]; then
+      wait_time=`expr $wait_time \* 2`
+    fi
   done
 
+  echo "$filename did not appear after 60 seconds."
   return 1
 }
 
